### PR TITLE
cleanup prints | have multicore configs show properly in collateral

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Type of change**: bug fix + other enhancement

**Impact**: no rtl change

**Development Phase**: implementation

**Release Notes**
This adds a new function called `AddToStringPrefix` that makes it so that the prefix for all the BOOM prints is in one location. Additionally, there is a couple of new print headers and spaces to make things a bit clearer. Finally, the `*.core.config` file that is generated is made in the subsystem level so that it can include all tiles in the system rather than just the first.

### CODE REVIEW

File: src/main/scala/util/util.scala [hunk 1]
```diff
@@ -572,3 +574,17 @@ object FPRegToChars
     multiVec(fpreg)
   }
 }
+
+object AddToStringPrefix
```
Comment (bkorpan) [inline src/main/scala/util/util.scala]: Would a more specific name like "BuildCoreInfoString" make sense?

File: src/main/scala/ifu/frontend.scala [hunk 2]
```diff
@@ -320,7 +321,12 @@ class BoomFrontendModule(outer: BoomFrontend) extends LazyModuleImp(outer)
   def ccover(cond: Bool, label: String, desc: String)(implicit sourceInfo: SourceInfo) =
     cover(cond, s"FRONTEND_$label", "Rocket;;" + desc)
 
-  override def toString: String = bpdpipeline.toString + "\n" + icache.toString
+  override def toString: String =
+    (AddToStringPrefix("====Overall Frontend Params====")
+    + "\n"
+    + bpdpipeline.toString
+    + "\n"
+    + icache.toString)
```
Comment (bkorpan) [inline src/main/scala/ifu/frontend.scala:327]: Not a fan of having "\n" on their own line.

Comment (jerryz123) [reply src/main/scala/ifu/frontend.scala:327]: I agree

File: src/main/scala/util/util.scala (hunk 1 above)
Comment (abejgonzalez) [reply src/main/scala/util/util.scala]: I agree the function could change. Maybe `BoomCoreStringPrefix` also? Do you like `prefix` in the name?

File: src/main/scala/ifu/frontend.scala (hunk 2 above)
Comment (abejgonzalez) [reply src/main/scala/ifu/frontend.scala:327]: I can change that.

File: src/main/scala/util/util.scala (hunk 1 above)
Comment (bkorpan) [reply src/main/scala/util/util.scala]: Sure

**Review by bkorpan — APPROVED** [review]

**Pull request merged**